lib/parser/message.h

```diff
@@ -84,8 +88,14 @@ class MessageFormattedText {
   std::string MoveString() { return std::move(string_); }
 
 private:
-  std::string string_;
+  void Format(MessageFixedText text, ...);
+  template<typename A> A Convert(A &&x) { return std::forward<A>(x); }
+  const char *Convert(const std::string &x) { return x.c_str(); }
```

Comment:
I tried to see if this PR is fixing #446. Unfortunately, it is not. The issue is that when getting `MessageFormattedText{text, s}` where s is a `std::string`  that is not const, the compiler instantiates the template version with `A=std::string&` rather than selecting the `const std::string&` version as it would have done if there was no template version.... Sight, mixing templates and overloads is amazingly complex.
Three solutions I see:
- add other overloads (empirical and endless fight IMHO)
- make it only one function with `auto` return type and `if constexpr` on types
- add an SFINAE barrier on the template version

I favor the last one (tested to fix #446 with clang-7.1 and its libc++):
```
using DisableForTypes = std::enable_if_t<(... && !std::is_same_v<A, B>)>;
  template<typename A,
      typename = DisableForTypes<std::decay_t<A>, std::string, CharBlock>>
  A Convert(A &&x) {
    return std::forward<A>(x);
  }
```